zipkin-courrelate/src/main/scala/org/coursera/zipkin/courrelate/Main.scala

```diff
@@ -0,0 +1,139 @@
+package org.coursera.zipkin.courrelate
+
+import com.google.common.net.InetAddresses
+import com.twitter.logging.Logger
+import com.twitter.scrooge.BinaryThriftStructSerializer
+import com.twitter.util.Base64StringEncoder
+import com.twitter.server.{Closer, TwitterServer}
+import com.twitter.util.{Await, Closable, Future}
+import com.twitter.zipkin.receiver.kafka.KafkaProcessor
+import com.twitter.zipkin.receiver.kafka.KafkaSpanReceiverFactory
+import com.twitter.zipkin.thriftscala.Constants
+import com.twitter.zipkin.{thriftscala => thrift}
+import org.apache.log4j.ConsoleAppender
+import org.apache.log4j.Level
+import org.apache.log4j.PatternLayout
+import scala.util.Try
+
+object StringDecoder extends KafkaProcessor.KafkaDecoder {
+  private[this] val deserializer = new BinaryThriftStructSerializer[thrift.Span] {
+    override val encoder = Base64StringEncoder
+    val codec = thrift.Span
+  }
+
+  override def fromBytes(
+    bytes: Array[Byte]): Option[List[thrift.Span]] = {
+
+    val s = new String(bytes)
+    Some(s.split(",").flatMap(d => Try(deserializer.fromString(d)).toOption).toList)
+  }
+}
+
+object Main extends TwitterServer with KafkaSpanReceiverFactory with Closer {
+  // initialize logging (Kafka uses log4j)
+  val logger = Logger.get("Main")
+  val root = org.apache.log4j.Logger.getRootLogger()
+  root.addAppender(new ConsoleAppender(new PatternLayout("%r [%t] %p %c %x - %m%n")))
+  root.setLevel(Level.INFO)
+  
+
+  import SpanUtils._
+
+  val producer = new CourrelateProducer()
+
+
+  def main(): Unit = {
+    var (totalReceived, totalServerReceived, totalProcessed) = (0, 0, 0)
```

Comment:
use atomics here? else you'll run into race conditions when you have multiple consumer threads